desktop-widgets/printerwebengine.cpp

```diff
@@ -0,0 +1,148 @@
+// SPDX-License-Identifier: GPL-2.0
+#include "printerwebengine.h"
+#include "mainwindow.h"
+#include "templatelayout.h"
+#include "core/statistics.h"
+#include "core/qthelper.h"
+#include "core/settings/qPrefDisplay.h"
+#include <stdio.h>
+
+#include <algorithm>
+#include <QPainter>
+#include <QPrinter>
+#include <QtWebEngineWidgets>
+#include "profile-widget/profilewidget2.h"
+
+extern void exportProfile(const struct dive *dive, const QString filename);
+
+Printer::Printer(QPaintDevice *paintDevice, print_options &printOptions, template_options &templateOptions, PrintMode printMode, bool inPlanner, QWidget *parent = Q_NULLPTR) :
+	paintDevice(paintDevice),
+	printOptions(printOptions),
+	templateOptions(templateOptions),
+	printMode(printMode),
+	inPlanner(inPlanner),
+	done(0)
+{
+	webView = new QWebEngineView(parent);
+	connect(webView, &QWebEngineView::loadFinished, this, &Printer::onLoadFinished);
+		if (printMode == PRINT) {
+		connect(this, &Printer::profiles_inserted, this, &Printer::printing);
+	}
+	profiles_missing = true;
+}
+
+Printer::~Printer()
+{
+	delete webView;
+}
+
+void Printer::onLoadFinished()
+{
+	if (profiles_missing)
+		webView->page()->runJavaScript("   var profiles = document.getElementsByClassName(\"diveProfile\");\
+				       for (let profile of profiles) { \
+					  var id = profile.attributes.getNamedItem(\"Id\").value; \
+					  var img = document.createElement(\"img\"); \
+					  img.src = id + \".png\"; \
+					  img.style.height = \"100%\"; \
+					  img.style.width = \"100%\"; \
+					  profile.appendChild(img); \
+					} \
+				", [this](const QVariant &v) { emit profiles_inserted(); });
+	profiles_missing = false;
+	emit(progessUpdated(100));
+}
+
+void Printer::printing()
+{
+	QPrintDialog printDialog(&printer, (QWidget *) nullptr);
+	if (printDialog.exec() == QDialog::Accepted) {
+		webView->page()->print(&printer, [this](bool ok){ if (ok) emit jobDone(); });
+	}
```

Comment:
```suggestion
	if (printDialog.exec() == QDialog::Accepted)
		webView->page()->print(&printer, [this](bool ok){ if (ok) emit jobDone(); });
```

still not a fan of the C++ lambdas, i must admit.

Reply:
Done